File: Day 3/1.java
//SELF PALAK

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i,j;
        int size=nums.length;
        
        for(i=0;i<size-1;i++)
        {
            for(j=i+1;j<size;j++){
            if(nums[i]+nums[j]==target)
            {
                return new int[]{i, j};
            }
        }
        }
        return null;
    }
}

File: Day 3/1.cpp
//SELF KIN

class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<pair<int,int>> p;
        for(int i=0;i<nums.size();i++)
            p.push_back(make_pair(nums[i],i));
        sort(p.begin(),p.end());
        int st=0,end=p.size()-1;
        while(true)
        {
            if(p[st].first+p[end].first==target)
            {
                if(p[st].second==p[end].second)
                    return vector<int>{p[st].second,p[end].second};
                else
                    return vector<int>{p[st].second,p[end].second};
            }
            else if(p[st].first+p[end].first<target)
                st++;
            else
                end--;
        }
    }
